build.gradle.kts

```diff
@@ -18,197 +14,296 @@ import org.gradle.api.tasks.testing.logging.TestExceptionFormat.FULL
+    group = "release"
+    dependsOn(tasks.withType<GenerateMavenPom>())
+}
 
-    doLast {
-      maven {
-        pom {
-          project {
-            withGroovyBuilder {
-              "name"("App Engine Gradle Plugin")
-              "description"("This Gradle plugin provides tasks to build and deploy Google App Engine applications.")
+val tempRepoPath = "$buildDir/repo"
 
-              "url"("https://github.com/GoogleCloudPlatform/app-gradle-plugin")
-              "inceptionYear"("2016")
+val cleanTempRepo by tasks.registering(Delete::class) {
+    delete(tempRepoPath)
+}
 
-              "scm" {
-                "url"("https://github.com/GoogleCloudPlatform/app-gradle-plugin")
-                "connection"("scm:https://github.com/GoogleCloudPlatform/app-gradle-plugin.git")
-                "developerConnection"("scm:git://github.com/GoogleCloudPlatform/app-gradle-plugin.git")
-              }
+tasks.withType<PublishToMavenRepository>().configureEach {
+    dependsOn(cleanTempRepo)
+}
 
-              "licenses" {
-                "license" {
-                  "name"("The Apache Software License, Version 2.0")
-                  "url"("http://www.apache.org/licenses/LICENSE-2.0.txt")
-                  "distribution"("repo")
+publishing {
+    // `./gradlew publish` would generate the release artifacts to $buildDir/repo folder
+    repositories {
+        maven {
+            url = uri(tempRepoPath)
+        }
+    }
+    publications {
+        withType<MavenPublication> {
+            // Use the resolved versions in pom.xml
+            // Gradle might have different resolution rules, so we set the versions
+            // that were used in Gradle build/test.
+            versionMapping {
+                usage(Usage.JAVA_RUNTIME) {
+                    fromResolutionResult()
+                }
+                usage(Usage.JAVA_API) {
+                    fromResolutionOf("runtimeClasspath")
+                }
+            }
+            // Skip name and description for plugin markers
+            if (name == "pluginMaven") {
+                pom {
+                    name.set("App Engine Gradle Plugin")
+                    description.set("This Gradle plugin provides tasks to build and deploy Google App Engine applications.")
+                }
+            }
+            pom {
+                // <editor-fold defaultstate="collapsed" desc="Default POM values">
+                developers {
+                    developer {
+                        id.set("loosebazooka")
+                        name.set("Appu Goundan")
+                        email.set("appu@google.com")
+                    }
+                }
+                url.set("https://github.com/GoogleCloudPlatform/app-gradle-plugin")
+                inceptionYear.set("2016")
+                licenses {
+                    license {
+                        name.set("The Apache License, Version 2.0")
```

Comment:
Are you sure you want `The Apache Software License, Version 2.0` here?

Note that the official license text at https://www.apache.org/licenses/LICENSE-2.0.txt does not mention "software license", it has title of "Apache License Version 2.0" (without software).
The same goes for the license headers in the project: https://github.com/GoogleCloudPlatform/app-gradle-plugin/blob/f50be8dcdde14ec6ef94947f1e21f5ea855be792/src/main/java/com/google/cloud/tools/gradle/appengine/AppEnginePlugin.java#L4
They mention `Licensed under the Apache License, Version 2.0` rather than `Apache Software License`.

SPDX references the license as `Apache License 2.0` rather than `Apache Software License 2.0`: https://spdx.org/licenses/Apache-2.0.html

Reply:
I was trying to avoid changing the existing state as much as possible, but you are right.